testing/agent/src/main/java/sbt/ForkMain.java

```diff
@@ -294,7 +295,7 @@ private void runTestTasks(final ExecutorService executor, final Task[] tasks, fi
 					Task[] nestedTasks;
 					final TaskDef taskDef = task.taskDef();
 					try {
-						final List<ForkEvent> eventList = new ArrayList<ForkEvent>();
```

Comment:
Alternatively:

```
final Collection<ForkEvent> eventList = new java.util.concurrent.ConcurrentLinkedDeque<ForkEvent>();
```

Reply:
I like this if for no other reason that it makes it very explicit that we expect that the eventList may be touched by multiple threads.